src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -79,60 +79,86 @@ def parsePileupConfig(mcPileup, dataPileup):
 
 
 def _validateArgument(argument, value, argumentDefinition):
-    dictArguments = ['AcquisitionEra', 'ProcessingString', 'ProcessingVersion', 'MaxRSS', 'MaxVSize']
-    validNull = argumentDefinition[argument]["null"]
+    """
+    Validate a single argument against its definition in the spec
+    """
+    validNull = argumentDefinition["null"]
     if not validNull and value is None:
         raise WMSpecFactoryException("Argument %s can't be None" % argument)
     elif value is None:
         return value
 
     try:
-        value = argumentDefinition[argument]["type"](value)
+        value = argumentDefinition["type"](value)
     except Exception:
         raise WMSpecFactoryException("Argument: %s: value: %s type is incorrect in schema." % (argument, value))
 
-    validateFunction = argumentDefinition[argument]["validate"]
-    if validateFunction is not None:
-        try:
-            if argument in dictArguments and isinstance(value, dict):
-                validateArgumentDict(argument, value, validateFunction)
-            elif not validateFunction(value):
-                raise WMSpecFactoryException(
-                    "Argument %s: value: %s doesn't pass the validation function." % (argument, value))
-        except Exception as ex:
-            # Some validation functions (e.g. Lexicon) will raise errors instead of returning False
-            logging.error(str(ex))
-            raise WMSpecFactoryException("Validation failed: %s value: %s" % (argument, value))
+    _validateArgFunction(argument, value, argumentDefinition["validate"])
     return value
 
 
-def validateArgumentDict(argument, argValues, valFunc):
+def _validateArgumentDict(argument, argValue, argumentDefinition):
     """
     Validate arguments that carry a dict value type
     """
-    for value in argValues.values():
-        if not valFunc(value):
-            raise WMSpecFactoryException(
-                    "Argument %s: value: %s doesn't pass the validation function." % (argument, value))
+    validNull = argumentDefinition["null"]
+    if not validNull and None in argValue.values():
+        raise WMSpecFactoryException("Argument %s can't be None" % argument)
+    elif all(val is None for val in argValue.values()):
+        return argValue
+
+    for val in argValue.values():
+        try:
+            # sigh.. LumiList has a peculiar type validation
```

Comment:
this `LumiList` special handling is bad, but I'm out of ideas on how to make it better... :(